core/embed/rust/src/ui/component/model_t1/button.rs

```diff
@@ -0,0 +1,233 @@
+use crate::ui::{
+    component::{
+        model::{HidEvent, T1Button},
+        Component, Event, EventCtx,
+    },
+    display::{self, Color, Font},
+    geometry::{Offset, Point, Rect},
+};
+
+use super::theme;
+
+pub enum ButtonMsg {
+    Clicked,
+}
+
+#[derive(Copy, Clone)]
+pub enum ButtonPos {
+    Left,
+    Right,
+}
+
+impl ButtonPos {
+    fn hit(&self, b: &T1Button) -> bool {
+        matches!(
+            (self, b),
+            (Self::Left, T1Button::Left) | (Self::Right, T1Button::Right)
+        )
+    }
+}
+
+pub struct Button<T> {
+    area: Rect,
+    pos: ButtonPos,
+    baseline: Point,
+    content: ButtonContent<T>,
+    styles: ButtonStyleSheet,
+    state: State,
+}
+
+impl<T: AsRef<[u8]>> Button<T> {
+    pub fn new(
+        area: Rect,
+        pos: ButtonPos,
+        content: ButtonContent<T>,
+        styles: ButtonStyleSheet,
+    ) -> Self {
+        let (area, baseline) = Self::placement(area, pos, &content, &styles);
+        Self {
+            area,
+            pos,
+            baseline,
+            content,
+            styles,
+            state: State::Initial,
+        }
+    }
+
+    pub fn with_text(area: Rect, pos: ButtonPos, text: T, styles: ButtonStyleSheet) -> Self {
+        Self::new(area, pos, ButtonContent::Text(text), styles)
+    }
+
+    pub fn with_icon(
+        area: Rect,
+        pos: ButtonPos,
+        image: &'static [u8],
+        styles: ButtonStyleSheet,
+    ) -> Self {
+        Self::new(area, pos, ButtonContent::Icon(image), styles)
+    }
+
+    pub fn enable(&mut self, ctx: &mut EventCtx) {
+        self.set(ctx, State::Initial)
+    }
+
+    pub fn disable(&mut self, ctx: &mut EventCtx) {
+        self.set(ctx, State::Disabled)
+    }
+
+    pub fn is_enabled(&self) -> bool {
+        matches!(
+            self.state,
+            State::Initial | State::Pressed | State::Released
+        )
+    }
+
+    pub fn is_disabled(&self) -> bool {
```

Comment:
thinking about it T1 design has no disabled buttons so we can probably remove it altogether

Reply:
I think helper methods like this help a bit with some logic (like the one in keyboards).

Reply:
It's not free in programmer cognitive load though. Not opposed to shorthands that do very similar but slightly different things, but in this case it's literally one `!` away from the other method. As a user of the API, it would make me stop and think if there's some subtle reason to use one or the other.

Reply:
I have an exactly opposite experience 🤷

Reply:
Dropped in 6c38a3c397ff37342f00a3b73100bafa3a2a8036.